c/src/sink_channel_filter.rs

```diff
@@ -0,0 +1,102 @@
+use crate::{FoxgloveChannelDescriptor, FoxgloveChannelMetadata, FoxgloveKeyValue, FoxgloveString};
+
+/// A filter for channels that can be used to subscribe to or unsubscribe from channels.
+///
+/// This can be used to omit one or more channels from a sink, but still log all channels to another
+/// sink in the same context. The callback should return false to disable logging of this channel.
+///
+/// This method is invoked from the client's main poll loop and must not block.
+#[derive(Clone)]
+pub(crate) struct SinkChannelFilterHandler {
```

Comment:
Naming: I'm a little confused as to why this is named what it is. If I understand correctly, this is a concrete implementation of the SinkChannelFilter trait. I think of a "handler" as something responding to an event. I'm wondering if there's a more concise name here? Maybe the trait could be `ChannelFilter` and this implementation is a `SinkChannelFilter`? Not sure, just thinking out loud.